protostar/starknet_gateway/gateway_facade_input_validation.py

```diff
@@ -0,0 +1,97 @@
+import json
+from typing import List, Dict, Any
+
+from starkware.cairo.lang.compiler.parser import parse_type
+from starkware.cairo.lang.compiler.ast.cairo_types import TypePointer, TypeCodeoffset
+
+from protostar.protostar_exception import ProtostarException
+
+from starknet_py.utils.data_transformer.data_transformer import CairoSerializer
+from starkware.starknet.public.abi_structs import identifier_manager_from_abi
+
+
+class InvalidInputException(ProtostarException):
+    pass
+
+
+class JSONParsingErrorException(ProtostarException):
+    pass
+
+
+AbiType = List[Dict[str, Any]]
+ConstructorType = Dict[str, Any]
+
+
+def get_type_size(abi: AbiType, typename: str):
+    if typename == "felt":
+        return 1
+    [type_definition] = [x for x in abi if x["name"] == typename]
+    return type_definition["size"]
+
+
+def get_constructor(abi: AbiType) -> ConstructorType:
+    candidates = [x for x in abi if x["type"] == "constructor"]
+    if not candidates:
+        return {"inputs": [], "outputs": []}
+    assert len(candidates) == 1
+    return candidates[0]
+
+
+def get_abi_from_json(compiled_contract_json: str) -> AbiType:
```

Comment:
Can we make separate module abi_utils for this? Doesn't such utils exist already?